content/shmem_barrier_all.tex

```diff
@@ -23,8 +23,9 @@
 \apidescription{   
     The \FUNC{shmem\_barrier\_all} routine registers the arrival of a \ac{PE} at
     a barrier. Barriers are a fast mechanism for synchronizing all \acp{PE} at
-    once.  This routine causes a \ac{PE} to suspend execution until all \acp{PE}
-    have called \FUNC{shmem\_barrier\_all}.  This routine must be used with
+    once.  This routine blocks the \ac{PE} until all \acp{PE} have called
+    \FUNC{shmem\_barrier\_all}. In a multithreaded \openshmem
+    program, only the calling thread is blocked. This routine must be used with
     \acp{PE} started by \FUNC{shmem\_init}.
```

Comment:
add "or shmem_init_thread"
or substitute "\acp{PE} that initialized the \openshmem library"
or remove altogether? Isn't that precision redundant with the stated need for a OpenSHMEM program to call shmem_init explicitly since 1.2?

Reply:
Beside this,  👍

Reply:
The sentence was not introduced in this ticket. But, this is a good fix and collectives chapter committee can handle this. Can you please make a note there ?

Reply:
The sentence in question will be deleted by the contexts proposal.  :)

Reply:
I created #119 for this edit.  How about those +1's?